### PR TITLE
Comments got feelings too

### PR DESCRIPTION
**Pull request by RazvanBudacaTechquarter**
Users can now see if comments are good or bad. Server takes a bit long to respond, approx. 11 seconds, since I implemented this feature.

### CODE REVIEW

File: portfolio/src/main/java/com/budiwebsite/model/Comment.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.budiwebsite.model;
+
+public class Comment {
```
Comment (amaurymedeiros) [inline portfolio/src/main/java/com/budiwebsite/model/Comment.java:7]: Add some documentation to the class explaining what it stores and when/how it's expected to be used

File: portfolio/src/main/java/com/budiwebsite/model/Comment.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  String message;
+  double score;
```
Comment (amaurymedeiros) [inline portfolio/src/main/java/com/budiwebsite/model/Comment.java]: These properties should be private and only accessed from the getters below.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -68,12 +76,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
-    String comment = getUserComment(request, "comment-input", "");
+    String message = getUserComment(request, "comment-input", "");
     long timestamp = System.currentTimeMillis();
 
+    Document doc =
+        Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    double score = sentiment.getScore();
+    languageService.close();
```
Comment (amaurymedeiros) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Extract that to a function (maybe `calculateScore`?) with some documentation and below you can do something like `commentEntity.setProperty(SCORE, calculateScore(...)) `

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -22,7 +22,7 @@ function getComments() {
         const commentsContainer = document.getElementById("comments-container");
         commentsContainer.innerHTML = '';
         for(const comment of comments) {
```
Comment (amaurymedeiros) [inline portfolio/src/main/webapp/script.js]: nit: sorry for missing that before, but add a space between `for` and `(`

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -31,8 +31,12 @@ function deleteComments() {
...
 
-function createListElement(text) {
+function createListElement(message, score) {
   const liElement = document.createElement('li');
-  liElement.innerText = text;
+  liElement.innerText = message + ' (' + decideMorality(score) + ')';
   return liElement;
+}
+
+function decideMorality(score) {
```
Comment (amaurymedeiros) [inline portfolio/src/main/webapp/script.js:46]: add a comment explaining what's the possible outcome moralities and how the decision is made (i.e. what does it mean to have a negative score?)

File: portfolio/src/main/java/com/budiwebsite/model/Comment.java (hunk 2 above)
Comment (jgaillard85) [inline portfolio/src/main/java/com/budiwebsite/model/Comment.java]: Make those final as well if you are not planning to have setters for them

File: portfolio/src/main/java/com/budiwebsite/model/Comment.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package com.budiwebsite.model;
+
+/**
+ * Class stores comments inputted with their user along with their morality rating
```
Comment (amaurymedeiros) [inline portfolio/src/main/java/com/budiwebsite/model/Comment.java]: nit: finish sentences with "."

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -88,4 +97,17 @@ private int getMaxComms(HttpServletRequest request) {
     int maxComms = Integer.parseInt(request.getParameter("maxComms"));
     return maxComms < 0 ? DEFAULT_MAX_COMMS : maxComms;
   }
+
+  /**
+   * Based on the core sentiment of the message sent, 
+   * the AI calculates the "score" of a comment
```
Comment (amaurymedeiros) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: finish sentences with "."

Comment (amaurymedeiros) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Also it's a good place to say what "score" means...

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -31,8 +31,18 @@ function deleteComments() {
...
   const liElement = document.createElement('li');
-  liElement.innerText = text;
+  liElement.innerText = message + ' (' + decideMorality(score) + ')';
   return liElement;
+}
+
+/**
+ * Variable score is calculated by an AI and it reflects how positive your comment was.
+ * Score can range from -1 to 1.
+ * A negative score means your comment was hateful or negative
```
Comment (amaurymedeiros) [inline portfolio/src/main/webapp/script.js]: nit: finish sentences with "."

**Review by amaurymedeiros — APPROVED** [review]
feel free to merge it after addressing the comments

**Pull request merged**